gtfs-realtime/spec/en/reference.md

```diff
@@ -340,9 +354,89 @@ Identification information for the vehicle performing the trip.
 
 | _**Field Name**_ | _**Type**_ | _**Cardinality**_ | _**Description**_ |
 |------------------|------------|-------------------|-------------------|
-| **id** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | Internal system identification of the vehicle. Should be **unique** per vehicle, and is used for tracking the vehicle as it proceeds through the system. This id should not be made visible to the end-user; for that purpose use the **label** field |
+| **id** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | Internal system identification of the vehicle. Should be **unique** per vehicle, and is used for tracking the vehicle as it proceeds through the system. This id should not be made visible to the end-user; for that purpose use the **label** field. |
 | **label** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | User visible label, i.e., something that must be shown to the passenger to help identify the correct vehicle. |
 | **license_plate** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | The license plate of the vehicle. |
+| **carriage_descriptor** | [CarriageDescriptor](#message-carriagedescriptor) | repeated | An ordered list of carriage information. |
```

Comment:
Yes, I think we need to add `direction_id` in `TripDescriptor` in order to help disambiguate the vehicle. Here's two ways that producers could model a vehicle with a dead-end stop in its journey:
1. Have separate `trip_id` in GTFS-static for the legs before and after the dead-end station. Carriage sequence will change (reverse) when the `trip_id` (in `TripDescriptor`) changes. Based on my own analysis of a small set of random feeds, I'm expecting this to be the most-used way.
2. If one `trip_id` is used for the journey, have separate `direction_id` (in trips.txt) in GTFS-static for the legs before and after the dead-end station. Carriage sequence will change (reverse) when the `direction_id` (in `TripDescriptor`) changes.

Does that make sense? What do you think?